com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/jclouds/providers/aliyun/blobstore/AliOSSBlobStore.java

```diff
@@ -0,0 +1,328 @@
+import org.jclouds.blobstore.util.BlobUtils;
+import org.jclouds.domain.Location;
+import org.jclouds.io.ContentMetadata;
+import org.jclouds.io.Payload;
+import org.jclouds.io.PayloadSlicer;
+import org.jclouds.location.suppliers.LocationsSupplier;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Singleton
+public class AliOSSBlobStore extends BaseBlobStore {
+
+    private final AliOSSApi aliOSSApi;
+
+    @Inject
+    protected AliOSSBlobStore(AliOSSApi aliOSSApi, BlobStoreContext context, BlobUtils blobUtils, Supplier<Location> defaultLocation,
+                              LocationsSupplier locations, PayloadSlicer slicer) {
+        super(context, blobUtils, defaultLocation, locations, slicer);
+        this.aliOSSApi = aliOSSApi;
+    }
+
+    @Override
+    public boolean containerExists(String container) {
+        return doOssOperation(oss -> oss.doesBucketExist(container));
+    }
+
+    @Override
+    public PageSet<? extends StorageMetadata> list(String container, ListContainerOptions options) {
+        return doOssOperation(oss -> {
+            ListObjectsRequest request = toListObjectRequest(container, options);
+            ObjectListing objectListing = oss.listObjects(request);
+            List<StorageMetadata> storageMetadataList = objectListing.getObjectSummaries()
+                                                                     .stream()
+                                                                     .map(ossObjectSummary -> toStorageMetadata(oss, container, ossObjectSummary))
+                                                                     .collect(Collectors.toList());
+            return new PageSetImpl<>(storageMetadataList, objectListing.getNextMarker());
+        });
+    }
+
+    @Override
+    public boolean blobExists(String container, String name) {
+        return doOssOperation(oss -> oss.doesObjectExist(container, name));
+    }
+
+    @Override
+    public String putBlob(String container, Blob blob) {
+        return doOssOperation(oss -> {
+            try {
+                ObjectMetadata objectMetadata = createObjectMetadataFromBlob(blob);
+                PutObjectRequest request = new PutObjectRequest(container, blob.getMetadata()
+                                                                               .getProviderId(), blob.getPayload()
+                                                                                                     .openStream(), objectMetadata);
+                PutObjectResult result = oss.putObject(request);
+                return result.getETag();
+            } catch (IOException e) {
+                throw new IllegalStateException(e);
```

Comment:
Throw SLException instead.

Reply:
I didn`t wanted to mix our domain with the Jclouds domain, because we plan to submit a pull request to apache repo in the near future. But for now, it is reasonable.